packages/ui/src/core/image-comparison-slider/image-comparison-slider.component.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { intrinsicComponent } from '@scaleflex/ui/utils/functions';
+import { Color } from '@scaleflex/ui/utils/types/palette';
+import ArrowChange from '@scaleflex/icons/arrow-change';
+import { lightPalette } from '@scaleflex/ui/theme/roots/palette';
+
+import { ImageComparisonSliderProps } from './image-comparison-slider.props';
+import Styled from './image-comparison-slider.styles';
+import { getHorizontalPosition } from './image-comparison-slider.utils';
+
+const ImageComparisonSlider = intrinsicComponent<ImageComparisonSliderProps, HTMLDivElement>(
+  (
+    {
+      leftImgSrc,
+      leftImgAlt,
+      rightImgSrc,
+      rightImgAlt,
+      width = '1000px',
+      height = '1000px',
+      handleColor = lightPalette[Color.BackgroundStateless],
+      iconSize = 10,
+      iconPadding = 10,
+      handleIcon = <ArrowChange color={lightPalette[Color.IconsPrimary]} />,
+      ...rest
+    }: ImageComparisonSliderProps,
+    ref
+  ): JSX.Element => {
+    const [isResizing, setIsResizing] = useState(false);
+    const topImageRef = useRef<HTMLImageElement | null>(null);
+    const handleRef = useRef<HTMLDivElement | null>(null);
+
+    const setPositioning = useCallback((cursorHorizontalPosition: number) => {
+      if (topImageRef.current && handleRef.current) {
+        const { left, width } = topImageRef.current.getBoundingClientRect();
```

Comment:
We should differentiate between the width here and the one we are using as a prop